### PR TITLE
Update doc and natspec

### PR DESCRIPTION
**Pull request by Jean-Grimal**
Fixes :
- https://cantina.xyz/code/8409a0ce-6c21-4cc9-8ef2-bd77ce7425af/findings/a29d3775-3516-4013-b78a-69d00f0dc43a
- https://cantina.xyz/code/8409a0ce-6c21-4cc9-8ef2-bd77ce7425af/findings/3442e8f8-d0a2-46bd-83a8-67208038e5f5
- https://cantina.xyz/code/8409a0ce-6c21-4cc9-8ef2-bd77ce7425af/findings/1d74afc9-25a1-4c51-a004-fbaa4ee8f3e3
- https://cantina.xyz/code/8409a0ce-6c21-4cc9-8ef2-bd77ce7425af/findings/47b33a14-c10e-41cf-921e-19236493856a
- https://cantina.xyz/code/8409a0ce-6c21-4cc9-8ef2-bd77ce7425af/findings/0ce3ec22-ab52-4e6a-afed-ab98a7bc1fde
- https://cantina.xyz/code/8409a0ce-6c21-4cc9-8ef2-bd77ce7425af/findings/6a478755-6285-42fe-b0e9-04eb43ba6f2f

### CODE REVIEW

**Review by MerlinEgalite — CHANGES_REQUESTED** [review]
Just one thing, else LGTM!

**Review by MerlinEgalite — APPROVED** [review]

File: src/MetaMorpho.sol [hunk 1]
```diff
@@ -339,9 +338,10 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
     /// @notice Sets the withdraw queue as a permutation of the previous one, although markets with both zero cap and
     /// zero vault's supply can be removed from the permutation.
     /// @notice This is the only entry point to disable a market.
-    /// @notice Removing a market requires the vault to have 0 supply on it; but anyone can supply on behalf of the
-    /// vault so the call to `sortWithdrawQueue` can be griefed by a frontrun. To circumvent this, the allocator can
-    /// simply bundle a reallocation that withdraws max from this market with a call to `sortWithdrawQueue`.
+    /// @notice Removing a market requires the vault to have 0 supply on it, or to have previously submitted a removal
+    /// for this market (with the function `submitMarketRemoval`); but anyone can supply on behalf of the vault so the
+    /// call to `updateWithdrawQueue` can be griefed by a frontrun. To circumvent this, the allocator can simply bundle
+    /// a reallocation that withdraws max from this market with a call to `updateWithdrawQueue`.
```
Comment (Rubilmax) [inline src/MetaMorpho.sol:344]:
```suggestion
    /// @notice Removing a market requires the vault to have 0 supply on it, or to have previously submitted a removal
    /// for this market (with the function `submitMarketRemoval`).
    /// @notice Warning: Anyone can supply on behalf of the vault so the call to `updateWithdrawQueue` that expects a market to be empty can be griefed by a front-run. To circumvent this, the allocator can simply bundle a reallocation that withdraws max from this market with a call to `updateWithdrawQueue`.
```

**Review by MerlinEgalite — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -48,8 +48,8 @@ It can:
 - Set the rewards recipient.
 - Increase the timelock.
 - [Timelocked] Decrease the timelock.
-- [Timelocked with no possible veto] Set the performance fee (capped to 50%).
 - [Timelocked] Set the guardian.
+- Set the performance fee (capped to 50%).
```
Comment (Rubilmax) [inline README.md:52]:
```suggestion
- Set the performance fee (capped at 50%).
```

**Review by Rubilmax — APPROVED** [review]

**Review by MerlinEgalite — APPROVED** [review]

**Review by Rubilmax — APPROVED** [review]

**Review by MerlinEgalite — APPROVED** [review]

**Review by Rubilmax — APPROVED** [review]

**Review by adhusson — APPROVED** [review]

File: src/interfaces/IMetaMorpho.sol [hunk 3]
```diff
@@ -95,8 +95,7 @@ interface IMetaMorphoBase {
     /// @notice Revokes the pending cap of the market defined by `id`.
     function revokePendingCap(Id id) external;
 
-    /// @notice Submits a forced market removal from the vault, potentially losing all funds supplied to the market.
-    /// @dev Warning: Submitting a forced removal will overwrite the timestamp at which the market will be removable.
+    /// @notice Submits a forced market removal from the vault, eventually losing all funds supplied to the market.
```
Comment (QGarchery) [inline src/interfaces/IMetaMorpho.sol:98]: I would not change `potentially` into `eventually`, the funds will not necessarily be lost if the timelock is reached and the market effectively removed: funds could possibly be recovered later with a reallocate if the market allows it

**Review by QGarchery — APPROVED** [review]

File: src/interfaces/IMetaMorpho.sol (hunk 3 above)
Comment (Rubilmax) [reply src/interfaces/IMetaMorpho.sol:98]: This will be addressed in #354 pls don't change this PR let's merge it

**Review by MathisGD — APPROVED** [review]

**Comment by StErMi** [comment]
Looks good

**Pull request merged**